sw/device/lib/testing/check.h

```diff
@@ -45,4 +46,28 @@
     }                                                     \
   } while (false)
 
+/**
+ * Compare `num_items` of `got` against `expected` buffer.
+ *
+ * Prints differences between `got` and `expected` before generating a test
+ * assertion if a mismatch is found.
+ *
+ * `sizeof(got[0])` is used to determine the size of each item in the buffer.
+ *
+ * @param got Input buffer.
+ * @param expected Buffer containing expected values.
+ * @param num_items Number of items to compare.
+ * @param ... arguments to a LOG_* macro, which are evaluated if the check
+ * fails.
+ */
+#define CHECK_BUFFER(got, expected, num_items, ...)                           \
+  do {                                                                        \
+    if (memcmp(got, expected, num_items * sizeof(got[0])) != 0) {             \
+      for (size_t i = 0; i < num_items; ++i) {                                \
+        LOG_INFO("[%d] got = 0x%x; expected = 0x%x", i, got[i], expected[i]); \
+      }                                                                       \
+      CHECK(false, __VA_ARGS__);                                              \
```

Comment:
Maybe we should have an argument like `result_` instead of using `CHECK` here directly to let the callers handle the mismatch as they want. This could be useful if a test wants to compare multiple buffers (would print all mismatches) and would also work better with the `EXECUTE_TEST` macro.

Reply:
Done